pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -120,6 +126,17 @@ func GetClusterName(ctx context.Context, hostname string) string {
 	return getClusterName(ctx, defaultClusterNameData, hostname)
 }
 
+// GetRFC1123CompliantClusterName returns a k8s cluster name if it exists, either directly specified or autodiscovered
+// Some kubernetes cluster-names (EKS,AKS) are not RFC1123 compliant, mostly due to an `_`.
+// This function replaces the invalid `_` with a valid `-`.
+func GetRFC1123CompliantClusterName(ctx context.Context, hostname string) string {
+	clusterName := getClusterName(ctx, defaultClusterNameData, hostname)
+	if strings.Contains(clusterName, "_") {
+		log.Warnf("clustername: %s contains `_`, replacing it with `-` to be hostName compliant", clusterName)
+	}
+	return strings.ReplaceAll(clusterName, "_", "-")
```

Comment:
You also mentioned cluster names that can end with an invalid character. Should we strip them here? Can cluster name start with an invalid character too?

Reply:
> can start with an invalid

More information can be read here in the RFC: https://docs.google.com/document/d/1qrf6d2FJzYsvOOnx8tYdjH8uZAftXN_BHS3lvJNYQds/edit#heading=h.qifjhqhrghjo

AKS: can have a `_`, but need to start/end with a letter/number
EKS: can have a `_`, can end with a `_` or `-`, but not start with it
GKE: is always RFC compliant. 


Yes, but the possibility of them are really low, so as of right now I thought it makes sense to just warn them. But you are right, I don't see any reason why not just cutting the suffix if it exists 👍

Reply:
In example I believe this would be a valid clustername on EKS: `dasdsa---___----___---`

Reply:
I added this here: https://github.com/DataDog/datadog-agent/pull/9279/commits/1b0aa72ec5dbd86bede4f2624649247b29cc4874 

Will scope in @clamoriniere  and @mfpierre  on their opinion on trimming it. 
I above commit I added that but let's discuss whether we want this. The more editing we do the higher the potential collision.

Reply:
As discussed locally we decided to omit the trimming